swupd/manifest.go

```diff
@@ -541,8 +541,71 @@ func linkDeltaPeersForPack(c *config, oldManifest, newManifest *Manifest) error
 		}
 	}
 
-	// TODO: Once rename logic is in. Add code to identify what extra DeltaPeers
-	// should be created based on rename flag in newManifest.
+	// TODO: At the moment swupd-client looks at the new manifest renames entries to make a
+	// decision whether it should get a delta or not. If we change that to look instead at the
+	// list of deltas in the pack, we could run the full rename detection logic between
+	// arbitrary pairs of manifests, i.e. for any pack combination, and possibly generating new
+	// deltas for that specific combination.
+
+	// Identify deltas that are relevant to be included based on the rename flag.
+	//
+	// First walk to find potential targets for a rename. We will generate only a single delta
+	// for each target.
+	targets := make(map[Hashval]*File)
+
+	for _, nf := range newManifest.Files {
```

Comment:
This is a complicated function and does a lot of walking of the manifest. With #154 a function is exposed to link old renames (`linkOldRenames`) that doesn't cost O(n<sup>2</sup>) to compute. 

In fact, now that `linkPeersAndChange` does all the linking even for old renames, does it make sense to going back to using that function?

Reply:
I might be missing something, but these additions did not cause the function to be O(N^2).

linkOldRenames sets up DeltaPeers between files of the same manifest, we are doing something different here, we are linking with files from the old manifest, because their version might differ from the version mentioned in new manifest (in rename). That scenario is when a "new delta" may happen.

I still prefer handling things separatedly from linkPeersAndChange, that function still mutates files in different conditions we don't want. I rather we get things working, and then try to settle some duplication.

Reply:
ok, right. Forgot this was between any two manifests, not previous/current.